### ISSUE TITLE
feature: Drop connection quality to POOR when participant is reconnecting

### ISSUE THREAD

**Issue by davidzhao** [livekit/livekit#2048]
When one participant is in reconnecting, it's not visible to the other users. We do not want to mark them disconnected preemptively, but it would be helpful to others in the same room that they are having issues connecting.

ConnectionQuality is a good way to communicate that. We can immediately flip connection quality to poor on signs of the connection struggling.

**Comment by davidzhao** [comment]
Another proposal is to add a new ConnectionQuality value `Disconnected` to signal that we have not received recent packets from the participant.

**Comment by davidzhao** [comment]
Using the lack of packet reception as a signal makes sense! Yeah let's start with Upstream only. If the user is not publishing anything, their disconnected state would be less relevant to others anyways.

**Comment by arnasledev** [comment]
is it possible for the socket connection to be closed (`PING_TIMEOUT` or `TRANSPORT_CLOSE`) which means the reconnecting state, but RTC Connection still be alive?

**Comment by boks1971** [comment]
Yeah, that is possible. As these are two separate connections (actually three, one for signal and two peer connections), it is possible that only one experiences issue. But, a signal connection disruption does not affect media flow (media can continue to operate normally). Also, a signal connection disruption is addressed using an ICE Restart which is usually seamless.
